docs/conf.py

```diff
@@ -271,12 +297,51 @@ def _parse_object_description_signature(
     return signature
 
 
+def _parse_confval_signature(
```

Comment:
I'm guessing this is another test to make sure an xref to a confval is resolved?

Reply:
Yeah it checks for that, but it also pulls in the type and default value, so that we no longer have to manually include that information in the description text.

Using a type like `Dict[str, str]` in `add_config_value` is a bit questionable since Sphinx doesn't check them properly (in contrast the object_description_options are properly type checked), but still they seem to work.